[aarch64][caffe2] Fix FBGEMM detection on aarch64

andrewjcg opened the pull request:
Summary: Update as per-comment: FBGEMM works on aarch64, just not on x86_32.

Test Plan: CI

Differential Revision: D88096853


submodule

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01

malfet commented on CMakeLists.txt:
```diff
@@ -900,9 +900,9 @@ include(ExternalProject)
+if(USE_FBGEMM AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "^(x86_64|aarch64)$")
   message(WARNING
-    "x64 operating system is required for FBGEMM. "
+    "64-bit operating system is required for FBGEMM. "
```
This is a wrong statement. This  will not work on RISCV or S390, will it? If one really wants to check for 64-bit OS, they need to modify the condition to `CMAKE_SIZEOF_VOID_P EQUALS 8`

malfet replied:
Looks good to me, thank you for the update

malfet reviewed (changes requested):
IMO this is a misleading statement and will lead to compilation errors on RiscV, PowerPC, S390 hardware. Also, PyTorch is not really supported on 32-bit OSes, so one can just remove this check completely

andrewjcg committed:
[aarch64][caffe2] Fix FBGEMM detection on aarch64 (#169379)

Summary:

Update as per-comment: FBGEMM works on aarch64, just not on x86_32.

Test Plan: CI

Differential Revision: D88096853

huydhn commented:
> IMO this is a misleading statement and will lead to compilation errors on RiscV, PowerPC, S390 hardware. Also, PyTorch is not really supported on 32-bit OSes, so one can just remove this check completely

@malfet Does this mean that we can just remove `AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "^(x86_64|aarch64)$"` and depend on `USE_FBGEMM`?  If yes, it does make the logic simpler

malfet commented:
@pytorchbot rebase -b main

huydhn committed:
Use clang21 to avoid the need to pull in https://github.com/pytorch/FBGEMM/commit/cf5dee2dc3870c36792e664cd44c6287e4d1edd8

huydhn committed:
Ignore -Wunneeded-internal-declaration on clang21

huydhn committed:
Revert the change in qembeddingbag.cpp

huydhn committed:
Go back to gcc13

huydhn committed:
Merge branch 'main' into export-D88096853

Signed-off-by: Huy Do <huydhn@gmail.com>

huydhn committed:
Bump FBGEMM commit to fetch https://github.com/pytorch/FBGEMM/commit/c246916f9e3804eacc3c95058e51cce02ae00fff

Signed-off-by: Huy Do <huydhn@gmail.com>

huydhn commented:
With https://github.com/pytorch/FBGEMM/pull/5207 landed, this should work now across all platforms
